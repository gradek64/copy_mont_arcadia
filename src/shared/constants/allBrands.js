module.exports = [
  'topshop',
  'burton',
  'wallis',
  'topman',
  'evans',
  'missselfridge',
  'dorothyperkins',
]
