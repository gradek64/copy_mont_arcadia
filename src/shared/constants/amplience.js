import PropTypes from 'prop-types'

export const VIDEO_FORMAT = 'mp4'
export const VIDEO_SIZE = '720p'

export const IMAGE_FORMAT = 'jpg'
export const IMAGE_SIZES_RANGE = [
  '75',
  '100',
  '150',
  '200',
  '300',
  '400',
  '500',
  '700',
  '1000',
  '1300',
  '2400',
]
export const PROGRESSIVE_JPG_PARAM = 'fmt.jpeg.interlaced'
export const RECOMMENDED_QUALITY_PARAM = 'qlt=80'

export const imageSizesPropTypes = PropTypes.shape({
  mobile: PropTypes.number.isRequired,
  tablet: PropTypes.number.isRequired,
  desktop: PropTypes.number.isRequired,
})

export const amplienceImagesPropTypes = PropTypes.arrayOf(
  PropTypes.string.isRequired
)

export const amplienceAssetsPropTypes = PropTypes.shape({
  images: amplienceImagesPropTypes,
  video: PropTypes.string,
})

export const amplienceAssetsDefaultProps = {
  images: [],
  video: '',
}

export const IMAGE_SIZES = {
  thumbnails: {
    mobile: 100,
    tablet: 100,
    desktop: 100,
  },
  overlayLarge: {
    mobile: 750,
    tablet: 710,
    desktop: 1060,
  },
  productMedia: {
    mobile: 750,
    tablet: 710,
    desktop: 1060,
  },
  bundleProductSlots: {
    mobile: 170,
    tablet: 170,
    desktop: 170,
  },
  smallProduct: {
    mobile: 140,
    tablet: 140,
    desktop: 140,
  },
  miniBag: {
    mobile: 90,
    tablet: 130,
    desktop: 130,
  },
  plp: {
    1: {
      mobile: 355,
      tablet: 355,
      desktop: 355,
    },
    2: {
      mobile: 175,
      tablet: 260,
      desktop: 475,
    },
    3: {
      mobile: 140,
      tablet: 255,
      desktop: 315,
    },
    4: {
      mobile: 125,
      tablet: 125,
      desktop: 230,
    },
  },
}
