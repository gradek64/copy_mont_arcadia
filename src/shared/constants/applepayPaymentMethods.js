import { VISA, MASTERCARD, AMEX, SWITCH } from './paymentTypes'

export default {
  [VISA]: 'visa',
  [MASTERCARD]: 'masterCard',
  [AMEX]: 'amex',
  [SWITCH]: 'maestro',
}
