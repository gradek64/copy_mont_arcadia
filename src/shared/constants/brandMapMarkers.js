const defaultData = {
  scale: 1,
  size: {
    width: 20,
    height: 20,
  },
  anchor: {
    x: 10,
    y: 10,
  },
}

export default {
  burton: defaultData,
  dorothyperkins: defaultData,
  evans: defaultData,
  missselfridge: defaultData,
  topshop: defaultData,
  topman: {
    scale: 0.5,
    size: {
      width: 27,
      height: 38,
    },
    anchor: {
      x: 13,
      y: 19,
    },
  },
  wallis: {
    scale: 0.66,
    size: {
      width: 20,
      height: 33,
    },
    anchor: {
      x: 14,
      y: 15,
    },
  },
}
