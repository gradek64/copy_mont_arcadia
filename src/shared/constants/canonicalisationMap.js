export default {
  'm.topshop.com': 'http://www.topshop.com',
  'm.de.topshop.com': 'http://de.topshop.com',
  'm.fr.topshop.com': 'http://fr.topshop.com',
  'm.us.topshop.com': 'http://us.topshop.com',
  'm.eu.topshop.com': 'http://eu.topshop.com',
  'm.burton.co.uk': 'http://www.burton.co.uk',
  'm.eu.burton-menswear.com': 'http://eu.burton-menswear.com',
  'm.dorothyperkins.com': 'http://www.dorothyperkins.com',
  'm.euro.dorothyperkins.com': 'http://euro.dorothyperkins.com',
  'm.us.dorothyperkins.com': 'http://us.dorothyperkins.com',
  'm.evans.co.uk': 'http://www.evans.co.uk',
  'm.evansmode.de': 'http://www.evansmode.de',
  'm.euro.evansfashion.com': 'http://euro.evansfashion.com',
  'm.evansusa.com': 'http://www.evansusa.com',
  'm.missselfridge.com': 'http://www.missselfridge.com',
  'm.missselfridge.de': 'http://www.missselfridge.de',
  'm.missselfridge.fr': 'http://www.missselfridge.fr',
  'm.euro.missselfridge.com': 'http://euro.missselfridge.com',
  'm.us.missselfridge.com': 'http://us.missselfridge.com',
  'm.topman.com': 'http://www.topman.com',
  'm.us.topman.com': 'http://us.topman.com',
  'm.de.topman.com': 'http://de.topman.com',
  'm.fr.topman.com': 'http://fr.topman.com',
  'm.eu.topman.com': 'http://eu.topman.com',
  'm.wallis.co.uk': 'http://www.wallis.co.uk',
  'm.wallismode.de': 'http://www.wallismode.de',
  'm.euro.wallisfashion.com': 'http://euro.wallisfashion.com',
  'm.wallisfashion.com': 'http://www.wallis.co.uk',
}
