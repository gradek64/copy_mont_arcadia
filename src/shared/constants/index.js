export { default as customerDetailsSchema } from './customerDetailsSchema'
export { default as paymentSchema } from './paymentSchema'
