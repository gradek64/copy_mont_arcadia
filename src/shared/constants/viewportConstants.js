export const PORTRAIT = 'portrait'
export const LANDSCAPE = 'landscape'

export const mobile = 'mobile'
export const tablet = 'tablet'
export const laptop = 'laptop'
export const desktop = 'desktop'
