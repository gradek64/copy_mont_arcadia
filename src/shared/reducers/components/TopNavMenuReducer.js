import createReducer from '../../lib/create-reducer'
import { tablet, laptop, desktop } from '../../constants/viewportConstants'

export const initialState = { open: false, scrollToTop: false }

export default createReducer(initialState, {
  TOGGLE_TOP_NAV_MENU: (state) => ({ ...state, open: !state.open }),
  CLOSE_TOP_NAV_MENU: (state) => ({ ...state, open: false }),
  TOGGLE_SCROLL_TO_TOP: (state) => ({
    ...state,
    scrollToTop: !state.scrollToTop,
  }),
  UPDATE_MEDIA_TYPE: (state, payload) => {
    if (payload === undefined) return state
    if ([tablet, laptop, desktop].includes(payload.media)) {
      return initialState
    }
    return state
  },
})
