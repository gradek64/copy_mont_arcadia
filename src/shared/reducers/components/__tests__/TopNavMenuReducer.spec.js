import { createStore } from 'redux'
import TopNavMenuReducer, {
  initialState,
} from '../../components/TopNavMenuReducer'
import { tablet, laptop } from '../../../constants/viewportConstants'

describe('TopNavMenuReducer', () => {
  const store = createStore(TopNavMenuReducer)

  it('Load initialState', () => {
    expect(store.getState()).toMatchSnapshot()
  })

  it('TOGGLE_TOP_NAV_MENU', () => {
    store.dispatch({ type: 'TOGGLE_TOP_NAV_MENU' })
    expect(store.getState()).toEqual({ open: true, scrollToTop: false })
  })

  it('CLOSE_TOP_NAV_MENU', () => {
    store.dispatch({ type: 'TOGGLE_TOP_NAV_MENU' })
    expect(store.getState()).toEqual(initialState)
  })

  it('TOGGLE_SCROLL_TO_TOP', () => {
    store.dispatch({ type: 'TOGGLE_TOP_NAV_MENU' })
    expect(store.getState()).toEqual({ open: true, scrollToTop: false })
  })

  it('UPDATE_MEDIA_TYPE', () => {
    store.dispatch({ type: 'CLOSE_TOP_NAV_MENU' })
    store.dispatch({ type: 'UPDATE_MEDIA_TYPE', payload: { media: laptop } })
    expect(store.getState()).toEqual(initialState)
    store.dispatch({ type: 'UPDATE_MEDIA_TYPE', payload: { media: tablet } })
    expect(store.getState()).toEqual(initialState)

    store.dispatch({ type: 'UPDATE_MEDIA_TYPE' })
    expect(store.getState()).toEqual(initialState)
  })
})
