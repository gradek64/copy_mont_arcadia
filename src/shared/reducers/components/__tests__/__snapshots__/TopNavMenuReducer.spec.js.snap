// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavMenuReducer Load initialState 1`] = `
Object {
  "open": false,
  "scrollToTop": false,
}
`;
